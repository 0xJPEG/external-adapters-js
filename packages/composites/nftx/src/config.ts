import { Config as BaseConfig } from '@chainlink/types'
import { Requester, util } from '@chainlink/ea-bootstrap'
import { ethers } from 'ethers'

export const ENV_VAULT_ADDRESS = 'VAULT_ADDRESS'
export const ENV_TOKEN_ADDRESS = 'TOKEN_ADDRESS'
export const ENV_ETHEREUM_RPC_URL = 'ETHEREUM_RPC_URL'
export const ENV_FALLBACK_RPC_URL = 'RPC_URL'

export const DEFAULT_VAULT_ADDRESS = '0x269616D549D7e8Eaa82DFb17028d0B212D11232A'
export const DEFAULT_TOKEN_ADDRESS = '0x9cea2eD9e47059260C97d697f82b8A14EfA61EA5'
export const DEFAULT_ENDPOINT = 'punk'
export const NAME = 'NFTX_VAULT_PRICE'

export type Config = BaseConfig & {
  vaultAddress: string
  tokenAddress: string
  provider: ethers.providers.Provider
}

export const makeConfig = (prefix?: string): Config => {
  const rpcUrl = util.getRequiredEnvWithFallback(
    ENV_ETHEREUM_RPC_URL,
    [ENV_FALLBACK_RPC_URL],
    prefix,
  )

  return {
    ...Requester.getDefaultConfig(prefix),
    vaultAddress: util.getEnv(ENV_VAULT_ADDRESS, prefix) || DEFAULT_VAULT_ADDRESS,
    tokenAddress: util.getEnv(ENV_TOKEN_ADDRESS) || DEFAULT_TOKEN_ADDRESS,
    provider: new ethers.providers.JsonRpcProvider(rpcUrl),
    defaultEndpoint: DEFAULT_ENDPOINT,
  }
}
