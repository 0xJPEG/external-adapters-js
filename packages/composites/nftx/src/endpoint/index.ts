export * as fees from './fees'
export * as price from './price'
