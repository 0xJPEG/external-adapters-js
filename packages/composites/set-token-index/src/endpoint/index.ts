export * as allocations from './allocations'
export * as tokens from './tokens'
export * as tvl from './tvl'
