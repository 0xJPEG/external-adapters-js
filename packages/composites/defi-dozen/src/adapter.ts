import { Validator } from '@chainlink/ea-bootstrap'
import { Execute, ExecuteWithConfig, Config } from '@chainlink/types'
import { makeConfig } from './config'
import * as TokenAllocation from '@chainlink/token-allocation-adapter'

const getSymbols = async (): Promise<any> => {
  const symbols = await import(`./symbols/symbols.json`)
  console.log(symbols)
  return symbols
}

export const execute: ExecuteWithConfig<Config> = async (input, context, config) => {
  const validator = new Validator(input)
  if (validator.error) throw validator.error

  const jobRunID = validator.validated.jobRunID
  const allocation = Object.keys(await getSymbols())
  const _execute = TokenAllocation.makeExecute()
  console.log(config)
  return await _execute({ id: jobRunID, data: { ...input.data, allocation } }, context)
}

export const makeExecute = (config?: Config): Execute => {
  return async (request, context) => execute(request, context, config || makeConfig())
}
