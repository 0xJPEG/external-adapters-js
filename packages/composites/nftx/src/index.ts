import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import { makeConfig } from './config'
import * as types from './endpoint'

const NAME = 'NFTX_VAULT_PRICE'

const { server } = expose(NAME, makeExecute(), undefined, endpointSelector)
export { NAME, makeExecute, makeConfig, server, types }
