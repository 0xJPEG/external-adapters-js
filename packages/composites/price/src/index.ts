import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './adapter'
import { makeConfig } from './config'

const NAME = 'ANCHOR_PROTOCOL'

export = { NAME, makeConfig, makeExecute, ...expose(NAME, makeExecute()) }
