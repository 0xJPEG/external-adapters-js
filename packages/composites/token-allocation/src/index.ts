import { expose } from '@chainlink/ea-bootstrap'
import { makeExecute } from './endpoint'
import { makeEndpointSelector } from './adapter'
import { makeConfig } from './config'
import * as types from './types'

const NAME = 'Token-Allocation'
const { server } = expose(NAME, makeExecute())

export { NAME, types, makeExecute, makeConfig, server, makeEndpointSelector }
