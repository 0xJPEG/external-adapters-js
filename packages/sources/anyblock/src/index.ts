import { expose } from '@chainlink/ea-bootstrap'
import { endpointSelector, makeExecute } from './adapter'
import * as endpoints from './endpoint'
import { makeConfig } from './config'

const NAME = 'ANYBLOCK'

const { server } = expose(NAME, makeExecute(), undefined, endpointSelector)

export { NAME, endpoints, makeExecute, makeConfig, server }
