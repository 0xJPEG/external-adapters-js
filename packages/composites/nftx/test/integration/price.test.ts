import { AdapterRequest } from '@chainlink/types'
import request, { SuperTest, Test } from 'supertest'
import process from 'process'
import nock from 'nock'
import http from 'http'
import { server as startServer } from '../../src'
import { mockEthereumResponseSuccess } from './fixtures'
import { AddressInfo } from 'net'

let oldEnv: NodeJS.ProcessEnv

beforeAll(() => {
  oldEnv = JSON.parse(JSON.stringify(process.env))
  process.env.ETHEREUM_RPC_URL = process.env.ETHEREUM_RPC_URL || 'http://localhost:8545'
  if (process.env.RECORD) {
    nock.recorder.rec()
  }
})

afterAll(() => {
  process.env = oldEnv
  if (process.env.RECORD) {
    nock.recorder.play()
  }

  nock.restore()
  nock.cleanAll()
  nock.enableNetConnect()
})

describe('execute', () => {
  const id = '1'
  let server: http.Server
  let req: SuperTest<Test>

  beforeAll(async () => {
    server = await startServer()
    req = request(`localhost:${(server.address() as AddressInfo).port}`)
  })

  afterAll((done) => {
    server.close(done)
  })

  describe('with vault address', () => {
    const data: AdapterRequest = {
      id,
      data: { address: '0x269616d549d7e8eaa82dfb17028d0b212d11232a' },
    }

    it('should return success', async () => {
      mockEthereumResponseSuccess()

      const response = await req
        .post('/')
        .send(data)
        .set('Accept', '*/*')
        .set('Content-Type', 'application/json')
        .expect('Content-Type', /json/)
        .expect(200)
      expect(response.body).toMatchSnapshot()
    })
  })
})
