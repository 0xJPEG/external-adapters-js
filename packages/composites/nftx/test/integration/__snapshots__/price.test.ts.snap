// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`execute with vault address should return success 1`] = `
Object {
  "data": Object {
    "fee": "0.02",
    "price": "66.592359631913713045",
    "priceWithFee": "67.924206824551987306",
  },
  "jobRunID": "1",
  "statusCode": 200,
}
`;
